R/arith.R

```diff
@@ -60,7 +62,22 @@ Ops.units <- function(e1, e2) {
     units(e2) <- units(e1) # convert before we can compare; errors if unconvertible
   }
 
-  if (mod) {
+  identical_units <-
+    inherits(e1, "units") && inherits(e2, "units") &&
+    identical(units(e1), units(e2))
+
+  inverse_units <-
+    inherits(e1, "units") && inherits(e2, "units") &&
+    identical(units(e1)$numerator, units(e2)$denominator) &&
+    identical(units(e1)$denominator, units(e2)$numerator)
```

Comment:
If you don't mind, let's take the opportunity for a small improvement, which is to call `inherits(e1, "units")` and `inherits(e2, "units")` just _once_ in this function and then reuse the result, because there are too many of these calls already throughout the function.

Reply:
I updated this.  It required also making one `if` into an `else if`.